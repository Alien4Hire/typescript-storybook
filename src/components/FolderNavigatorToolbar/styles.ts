import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
