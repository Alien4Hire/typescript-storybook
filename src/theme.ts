import { createTheme } from '@mui/material/styles';
import mlColors from './util/ml-cdl-colors';
import {
  computerSmallMonitorMax,
  mobileScreenMax,
  mobileScreenMin,
  tabletScreenMax,
} from './util/breakpoints';

declare module '@mui/material/Typography' {
  interface TypographyPropsVariantOverrides {
    title: true;
    textLarge: true;
    textBasic: true;
    textSmall: true;
    textSub: true;
    buttonMedium: true;
    buttonSmall: true;
    buttonLarge: true;
  }
}

//UX variants definitions can be found at https://www.figma.com/file/bEonWYX4lesmqlFY6MqWCt/Course-Assembly-MVP?node-id=5540%3A249648
//Custom created variant names for typographies

declare module '@mui/material/styles/createTypography' {
  interface Typography {
    formTitle?: React.CSSProperties;
    formSubtitle?: React.CSSProperties;
    appTitle?: React.CSSProperties;
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }

  // allow configuration using `createMuiTheme`
  interface TypographyOptions {
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }
}

declare module '@mui/material/styles' {
  interface Theme {}
  interface ThemeOptions {}
  interface Palette {
    specials: Palette['primary'];
    aqua: Palette['primary'];
    black: Palette['primary'];
    blue: Palette['primary'];
    gray: Palette['primary'];
    green: Palette['primary'];
    magenta: Palette['primary'];
    orange: Palette['primary'];
    purple: Palette['primary'];
    red: Palette['primary'];
    teal: Palette['primary'];
    white: Palette['primary'];
    yellow: Palette['primary'];
  }
  interface PaletteOptions {
    specials: PaletteOptions['primary'];
    aqua: PaletteOptions['primary'];
    black: PaletteOptions['primary'];
    blue: PaletteOptions['primary'];
    gray: PaletteOptions['primary'];
    green: PaletteOptions['primary'];
    magenta: PaletteOptions['primary'];
    orange: PaletteOptions['primary'];
    purple: PaletteOptions['primary'];
    red: PaletteOptions['primary'];
    teal: PaletteOptions['primary'];
    white: PaletteOptions['primary'];
    yellow: PaletteOptions['primary'];
  }

  interface PaletteColor {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }

  interface SimplePaletteColorOptions {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }
}

export const theme = createTheme({
  palette: {
    background: {
      default: mlColors.off_white,
      paper: mlColors.pure_white,
    },
    primary: {
      main: mlColors.aqua,
    },
    secondary: {
      main: mlColors.aqua_dark,
    },
    specials: {
      main: mlColors.macmillan_red,
      macmillanRed: mlColors.macmillan_red,
      fullPageDialogShadow: 'rgba(0, 0, 0, 0.5)',
    },
    aqua: {
      main: mlColors.aqua,
      veryLight: mlColors.aqua_very_light,
      light: mlColors.aqua_light,
      medium: mlColors.aqua_medium,
      dark: mlColors.aqua_dark,
      veryDark: mlColors.dark_navy,
    },
    black: {
      main: mlColors.black,
    },
    blue: {
      main: mlColors.blue,
      veryLight: mlColors.blue_very_light,
      light: mlColors.blue_light,
      medium: mlColors.blue_medium,
      dark: mlColors.blue_dark,
    },
    gray: {
      main: mlColors.gray,
      veryLight: mlColors.gray_very_light,
      light: mlColors.gray_light,
      medium: mlColors.gray_medium,
      dark: mlColors.gray_dark,
      veryDark: mlColors.gray_very_dark,
    },
    green: {
      main: mlColors.green,
      veryLight: mlColors.green_very_light,
      light: mlColors.green_light,
      medium: mlColors.green_medium,
      dark: mlColors.green_dark,
    },
    magenta: {
      main: mlColors.magenta,
      veryLight: mlColors.magenta_very_light,
      light: mlColors.magenta_light,
      medium: mlColors.magenta_medium,
      dark: mlColors.magenta_dark,
    },
    orange: {
      main: mlColors.orange,
      veryLight: mlColors.orange_very_light,
      light: mlColors.orange_light,
      medium: mlColors.orange_medium,
      dark: mlColors.orange_dark,
    },
    purple: {
      main: mlColors.purple,
      veryLight: mlColors.purple_very_light,
      light: mlColors.purple_light,
      medium: mlColors.purple_medium,
      dark: mlColors.purple_dark,
    },
    red: {
      main: mlColors.red,
      veryLight: mlColors.red_very_light,
      light: mlColors.red_light,
      medium: mlColors.red_medium,
      dark: mlColors.red_dark,
    },
    teal: {
      main: mlColors.teal,
      veryLight: mlColors.teal_very_light,
      light: mlColors.teal_light,
      medium: mlColors.teal_medium,
      dark: mlColors.teal_dark,
      veryDark: mlColors.teal_very_dark,
    },
    white: {
      main: mlColors.pure_white,
      offWhite: mlColors.off_white,
    },
    yellow: {
      main: mlColors.yellow,
      veryLight: mlColors.yellow_very_light,
      light: mlColors.yellow_light,
      medium: mlColors.yellow_medium,
      dark: mlColors.yellow_dark,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: mobileScreenMin,
      md: mobileScreenMax,
      lg: tabletScreenMax,
      xl: computerSmallMonitorMax,
    },
  },
  spacing: 8,
  typography: {
    fontSize: 14,
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      'Oxygen',
      'Ubuntu',
      'Cantarell',
      '"Fira Sans"',
      '"Droid Sans"',
      '"Helvetica Neue"',
      'Source Sans Pro',
      'sans-serif',
    ].join(','),
    allVariants: {
      fontWeight: 'normal',
    },
    button: {
      fontSize: 14,
      textTransform: 'uppercase',
    },
    h1: {
      color: mlColors.black,
      fontSize: 32,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h2: {
      color: mlColors.black,
      fontSize: 24,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h3: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h4: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    title: {
      color: mlColors.black,
      fontSize: 28,
      fontFamily: 'Source Sans Pro',
      fontWeight: 200,
      fontStyle: 'normal',
    },
    textLarge: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textBasic: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSmall: {
      color: mlColors.black,
      fontSize: 14,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSub: {
      color: mlColors.black,
      fontSize: 13,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonLarge: {
      color: mlColors.black,
      fontSize: 15,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonMedium: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonSmall: {
      color: mlColors.black,
      fontSize: 12,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
  },
  components: {
    MuiAlert: {
      styleOverrides: {
        filledError: {
          background: mlColors.red_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.red}`,
        },
        filledInfo: {
          background: mlColors.blue_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.blue}`,
        },
        filledSuccess: {
          background: mlColors.green_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.green}`,
        },
        filledWarning: {
          background: mlColors.yellow_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.yellow}`,
        },
      },
    },
    // MuiButton: {
    //   styleOverrides: {
    //     root: {
    //       fontSize: '1rem',
    //       textTransform: 'none',
    //       height: '35px',
    //       padding: '0 15px',
    //       margin: '5px',
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     textSecondary: {
    //       fontSize: '0.9375rem',
    //       height: '34px',
    //       lineHeight: '33px',
    //       padding: '0 15px',
    //       borderRadius: '3px',
    //       '-webkit-border-radius': '3px',
    //       '-ms-border-radius': '3px',
    //       '-moz-border-radius': '3px',
    //       textDecoration: 'none',
    //       color: '#00758e',
    //       cursor: 'pointer',
    //       background: 'transparent',
    //       border: '1px solid #00758e',
    //       overflow: 'hidden',
    //     },
    //     containedPrimary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     outlinedSecondary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #00758e',
    //       },
    //       '&:disabled': {
    //         color: '#989898',
    //         fill: '#c8c8c8',
    //         backgroundColor: '#f3f3f3',
    //       },
    //     },
    //   },
    // },
    MuiButtonBase: {
      defaultProps: {
        disableRipple: true, // No more ripple, on the whole application!
        focusRipple: false,
      },
    },
  },
});
