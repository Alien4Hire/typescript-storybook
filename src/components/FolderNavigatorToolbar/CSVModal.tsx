import React from 'react';

import { Typography, Box, Link } from '@mui/material';
import { useCSVReader, useCSVDownloader } from 'react-papaparse';

import generateUuid from '../../util/uuid';
import { ReactComponent as DownloadIcon } from '../../icons/download.svg';
import { ReactComponent as LinkIcon } from '../../icons/link.svg';
import { Modal } from '../Modal/Modal';
import {
  CsvModalBackdrop,
  UploadListItem,
  UploadButton,
  DownloadTemplateWrapper,
} from './styles';
import { theme } from '../../theme';

export type UploadT = {
  title: string;
  url: string;
};

export type CsvModalT = {
  isCSVModalOpen: boolean;
  onCSVModalClose: () => void;
  onAddActivities: () => void;
  uploadList: UploadT[];
  setUploadList: (uploadList: UploadT[]) => void;
};

const CSV_TEMPLATE_DATA = [
  {
    Title: 'Example Activity 1',
    URL: 'http://example-1.com',
  },
  {
    Title: 'Example Activity 2',
    URL: 'http://example-2.com',
  },
  {
    Title: 'Example Activity 3',
    URL: 'http://example-3.com',
  },
];

const CSVModal = ({
  isCSVModalOpen,
  onCSVModalClose,
  onAddActivities,
  uploadList,
  setUploadList,
}: CsvModalT) => {
  const { CSVReader } = useCSVReader();
  const { CSVDownloader } = useCSVDownloader();

  const handleUploadCSV = (results: string[][]) => {
    const formattedResult: UploadT[] = [];
    for (let index = 0; index < results.length; index++) {
      const element = results[index];
      if (
        element[0].toLocaleLowerCase() !== 'title' &&
        element[1].toLocaleLowerCase() !== 'url'
      ) {
        formattedResult.push({ title: element[0], url: element[1] });
      }
    }
    setUploadList(formattedResult);
  };

  const handleReuploadCSV = () => {
    setUploadList([]);
  };

  return (
    <Modal
      title={'Add URL activities'}
      open={isCSVModalOpen}
      actions={
        uploadList.length > 0
          ? [
              {
                label: 'Reupload CSV',
                variant: 'outlined',
                color: 'primary',
                onClick: () => handleReuploadCSV(),
              },
              {
                label: 'Add Activities',
                variant: 'contained',
                color: 'primary',
                onClick: () => onAddActivities(),
              },
            ]
          : []
      }
      onClose={() => onCSVModalClose()}
    >
      {uploadList.length > 0 ? (
        <Box sx={{ padding: 1 }}>
          <Typography
            sx={{ fontStyle: 'italic', fontWeight: 'bold', marginBottom: 2 }}
            variant="subtitle2"
          >
            Previewing {uploadList.length} activities
          </Typography>
          <Box sx={{ maxHeight: 220, overflowY: 'auto' }}>
            {uploadList.map((item) => (
              <UploadListItem theme={theme} key={generateUuid()}>
                <LinkIcon />
                <Link
                  underline="none"
                  target="_blank"
                  href={item.url}
                  rel="noreferrer"
                  sx={{ marginLeft: 1 }}
                >
                  {item.title}
                </Link>
              </UploadListItem>
            ))}
          </Box>
        </Box>
      ) : (
        <CSVReader
          onUploadAccepted={(results: any) => {
            handleUploadCSV(results.data);
          }}
        >
          {({ getRootProps }: any) => (
            <>
              <CsvModalBackdrop>
                <Typography variant="body1">
                  Upload a CSV of URLs and their titles.
                </Typography>
                <UploadButton
                  data-testid="upload-button"
                  variant="contained"
                  {...getRootProps()}
                >
                  Upload .csv
                </UploadButton>
                <Typography
                  variant="body2"
                  sx={{ marginTop: 1, color: theme.palette.gray.dark }}
                >
                  Accepted file types: csv
                </Typography>
                <CSVDownloader filename="template" data={CSV_TEMPLATE_DATA}>
                  <DownloadTemplateWrapper>
                    <DownloadIcon color={theme.palette.primary.main} />
                    <Typography
                      data-testid="download-link"
                      variant="body2"
                      sx={{
                        marginLeft: 1.5,
                        color: theme.palette.primary.main,
                      }}
                    >
                      Download template
                    </Typography>
                  </DownloadTemplateWrapper>
                </CSVDownloader>
              </CsvModalBackdrop>
            </>
          )}
        </CSVReader>
      )}
    </Modal>
  );
};

export default CSVModal;
