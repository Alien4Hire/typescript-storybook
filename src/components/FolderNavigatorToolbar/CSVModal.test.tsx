import { render, screen } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import '@testing-library/jest-dom/extend-expect';

import { theme } from '../../theme';
import CSVModal, { CsvModalT } from './CSVModal';
import '../../setupTests';

const baseProps = {
  isCSVModalOpen: true,
  onCSVModalClose: jest.fn(),
  onAddActivities: jest.fn(),
  uploadList: [],
  setUploadList: jest.fn(),
};

const renderComponent = (customProps: Partial<CsvModalT> = {}) => {
  const props = { ...baseProps, ...customProps };
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <CSVModal {...props} />
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

describe('<CSVModal />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should have a title', () => {
    renderComponent();
    expect(screen.getByText('Add URL activities')).toBeInTheDocument();
  });

  it('should show the upload button', async () => {
    renderComponent();
    const testId = 'upload-button';
    const uploadButton = screen.getByTestId(testId);
    expect(uploadButton).toBeInTheDocument();
  });

  it('should show the download link', async () => {
    renderComponent();
    const testId = 'download-link';
    const downloadLink = screen.getByTestId(testId);
    expect(downloadLink).toBeInTheDocument();
  });

  it('should show the preview items', async () => {
    const uploadList = [
      {
        title: 'Example Activity 1',
        url: 'http://example-1.com',
      },
      {
        title: 'Example Activity 2',
        url: 'http://example-2.com',
      },
      {
        title: 'Example Activity 3',
        url: 'http://example-3.com',
      },
    ];
    renderComponent({ uploadList });
    uploadList.forEach((item, index) => {
      const listItem = screen.getByText(item.title);
      expect(listItem).toBeInTheDocument();
    });
  });
});
