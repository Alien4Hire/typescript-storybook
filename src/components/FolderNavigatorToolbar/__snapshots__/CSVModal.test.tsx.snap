// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CSVModal /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c5::-moz-focus-inner {
  border-style: none;
}

.c5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c6 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c6:hover {
  background-color: rgba(0,0,0,0.04);
}

.c6.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

.c4 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.235;
  word-break: break-all;
}

.c9 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
}

.c12 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #686868;
  margin-top: 8px;
}

.c14 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.43;
  color: #00758e;
  margin-left: 12px;
}

.c2 {
  width: 400px;
  background-color: #ffffff;
  padding: 16px;
  padding-left: 0px;
  padding-right: 0px;
  padding-bottom: 0px;
  box-shadow: 0px 1px 3px rgba(0,0,0,0.2);
  padding-top: 0px;
}

.c10 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #fff;
  background-color: #00758e;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.c10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0,81,99);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c10:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c10.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c10.Mui-disabled {
  color: rgba(0,0,0,0.26);
  box-shadow: none;
  background-color: rgba(0,0,0,0.12);
}

.c0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0,0,0,0.5);
  -webkit-tap-highlight-color: transparent;
}

.c3 {
  padding: 12px 12px 12px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid #dddddd;
}

.c7 {
  padding: 20px;
  word-break: break-word;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f1f1f1;
  border-radius: 16px;
  padding: 32px 32px 70px;
}

.c11 {
  text-transform: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin-top: 16px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px;
  cursor: pointer;
}

@media print {
  .c5 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c6:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c10:hover {
    background-color: #00758e;
  }
}

<body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root c0"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="c1"
        open=""
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="c2 MuiBox-root"
        tabindex="-1"
      >
        <div
          class="c3"
        >
          <h4
            class="c4 MuiTypography-root MuiTypography-h4"
          >
            Add URL activities
          </h4>
          <button
            class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium"
            tabindex="0"
            type="button"
          >
            <svg>
              x.svg
            </svg>
          </button>
        </div>
        <div
          class="c7"
        >
          <input
            accept="text/csv, .csv, application/vnd.ms-excel"
            autocomplete="off"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div
            class="c8 MuiBox-root"
          >
            <p
              class="c9 MuiTypography-root MuiTypography-body1"
            >
              Upload a CSV of URLs and their titles.
            </p>
            <button
              class="c5 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root c10 c11"
              data-testid="upload-button"
              tabindex="0"
              type="button"
            >
              Upload .csv
            </button>
            <p
              class="c12 MuiTypography-root MuiTypography-body2"
            >
              Accepted file types: csv
            </p>
            <a
              class=""
            >
              <div
                class="c13 MuiBox-root"
              >
                <svg
                  color="#00758e"
                >
                  download.svg
                </svg>
                <p
                  class="c14 MuiTypography-root MuiTypography-body2"
                  data-testid="download-link"
                >
                  Download template
                </p>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": @media print {

}

@media (hover:none) {

}

@media (hover:none) {

}

<div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
